### PR TITLE
Remove usage of prop-types from more components

### PR DESCRIPTION
**Pull request by craigpalermo**

### CODE REVIEW

File: packages/core/src/Swatch/Swatch.tsx [hunk 1]
```diff
@@ -15,25 +14,26 @@ const SwatchColor = styled.div`
...
-      />
-    ))}
-  </Flex>
-)
+export function Swatch({ colors, onClick, ...props }: SwatchProps): React.ReactElement {
+  return (
+    <Flex flexWrap='wrap' {...props}>
+      {colors.map((color, idx) => (
+        <SwatchColor
+          data-testid={`${idx}-${color}`}
```
Comment (sdalonzo) [inline packages/core/src/Swatch/Swatch.tsx:27]: n-b id in key

File: packages/core/src/Flex/Flex.tsx [hunk 2]
```diff
@@ -31,17 +31,16 @@ const flexPropTypes = {
...
+export type FlexProps = BoxProps &
+  SpaceProps &
+  WidthProps &
+  AlignItemsProps &
+  JustifyContentProps &
+  AlignContentProps &
+  FlexWrapProps &
+  FlexDirectionProps
+
+const Flex: React.FC<FlexProps> = styled(Box).attrs(({ wrap, align, justify, ...props }) => ({
```
Comment (craigpalermo) [inline packages/core/src/Flex/Flex.tsx:43]: change to `function`

File: packages/core/src/Step/Step.tsx [hunk 3]
```diff
@@ -20,23 +19,15 @@ const StepButton = styled(Button)`
...
+export type StepProps = {
   className?: string
   active?: boolean
   completed?: boolean
   children?: React.ReactNode | string
   onClick?: (unknown) => unknown
 }
 
-const Step: React.FC<IStepProps> = ({ className, active, completed, children, onClick, ...props }) => {
+export function Step({ className, active, completed, children, onClick, ...props }: StepProps) {
```
Comment (craigpalermo) [inline packages/core/src/Step/Step.tsx:30]: return type is missing

File: packages/core/src/ToastProvider/ToastProvider.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import type { IToastProps } from '../Toast'
+import type { ToastProps } from '../Toast'
```
Comment (sdalonzo) [inline packages/core/src/ToastProvider/ToastProvider.tsx:2]: n-b (next PR is fine) could collapse this linebreak

File: packages/core/src/ToggleBadge/ToggleBadge.spec.tsx [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import React from 'react'
-import { ToggleBadge, theme } from '..'
+import { ToggleBadge, createTheme } from '..'
```
Comment (sdalonzo) [inline packages/core/src/ToggleBadge/ToggleBadge.spec.tsx:2]: barrel trap!

File: packages/core/src/ToggleBadge/index.ts [hunk 6]
```diff
@@ -1 +1 @@
-export { default as ToggleBadge } from './ToggleBadge'
+export { ToggleBadge } from './ToggleBadge'
```
Comment (sdalonzo) [inline packages/core/src/ToggleBadge/index.ts:1]:
```suggestion
export { ToggleBadge, type ToggleBadgeProps } from './ToggleBadge'
```

File: packages/core/src/Tooltip/Tooltip.tsx [hunk 7]
```diff
@@ -126,7 +126,8 @@ const Tooltip: React.FC<InferProps<typeof propTypes>> = ({ children, ...props })
   )
 }
 
-Tooltip.propTypes = propTypes
-Tooltip.defaultProps = defaultProps
+_Tooltip.defaultProps = defaultProps
+
+const Tooltip = withTheme(_Tooltip)
```
Comment (sdalonzo) [inline packages/core/src/Tooltip/Tooltip.tsx:131]: n-b not sure withTheme is needed

File: packages/core/src/Tooltip/index.ts [hunk 8]
```diff
@@ -1 +1 @@
-export { default as Tooltip } from './Tooltip'
+export { Tooltip } from './Tooltip'
```
Comment (sdalonzo) [inline packages/core/src/Tooltip/index.ts:1]:
```suggestion
export { Tooltip, type TooltipProps } from './Tooltip'
```

**Review by sdalonzo — APPROVED** [review]

File: packages/core/src/Tooltip/Tooltip.tsx (hunk 7 above)
Comment (craigpalermo) [reply packages/core/src/Tooltip/Tooltip.tsx:131]: Left it there just in case. Worth looking at our usage of `withTheme` everywhere at some point

**Pull request merged**